network/containerizer/bridgepolicy.go

```diff
@@ -58,8 +60,14 @@ func NewBridgePolicy(cfgGetter environs.ConfigGetter, st SpaceBacking) (*BridgeP
 		return nil, errors.Annotate(err, "getting space name to space ID map")
 	}
 
+	spacesNames, err := st.SpaceNamesByID()
```

Comment:
We are hitting the DB multiple times to get the same data in a different "shape" at least here, but I seem to recall in other places too.

Given the emerging ubiquity of space lookups, and the fact that we have 3 state package methods for the purpose, I think we should do the following:
- Remove `SpaceNamesByID`, `SpaceIDsByName`, `SpaceInfosByID`.
- Instead have a single method to return all spaces as `network.SpaceInfos`.
- Move the lookup generations to the `SpaceInfos` type.

Then we only ever hit the DB once to materialise a the set of spaces, and the reshaping for lookups is done in-memory.

Reply:
Will take a look at this for a follow on.  Here an in other places.